src/compiler/checker.ts

```diff
@@ -20310,7 +20310,8 @@ namespace ts {
                     shouldSkipElaboration = true; // Retain top-level error for interface implementing issues, otherwise omit it
                 }
                 if (props.length === 1) {
-                    const propName = symbolToString(unmatchedProperty);
+                    const nameType = getSymbolLinks(unmatchedProperty).nameType;
+                    const propName = nameType && nameType.flags & TypeFlags.EnumLiteral ? `[${symbolToString(nameType.symbol, nameType.symbol.valueDeclaration)}]` : symbolToString(unmatchedProperty);

```

Comment:
Does it make sense to do this sort of work in `symbolToString` itself?

Reply:
It seems like it... grepping through our errors, we have existing errors like:

```
tests\baselines\reference\computedPropertyNames45_ES5.errors.txt
1:tests/cases/conformance/es6/computedProperties/computedPropertyNames45_ES5.ts(10,5): error TS2411: Property '["get1"]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
2:tests/cases/conformance/es6/computedProperties/computedPropertyNames45_ES5.ts(11,9): error TS2411: Property '["set1"]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
17:!!! error TS2411: Property '["get1"]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
21:!!! error TS2411: Property '["set1"]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
```

```
tests\baselines\reference\computedPropertyNames38_ES5.errors.txt
1:tests/cases/conformance/es6/computedProperties/computedPropertyNames38_ES5.ts(8,9): error TS2411: Property '[1 << 6]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
14:!!! error TS2411: Property '[1 << 6]' of type 'Foo' is not assignable to 'string' index type 'Foo2'.
```

So, surely somewhere else we are adding the `[]`, right?